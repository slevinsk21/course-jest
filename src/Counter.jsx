// react
import { useState } from 'react';
import { number } from 'prop-types';

const Counter = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleCounter = value => {
    value >= 0 && setCounter(value);
  };

  return (
    <>
      <div style={{ height: 60 }}></div>
      <div
        style={{
          backgroundColor: '#d3d3',
          borderRadius: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          margin: '0 auto',
          maxWidth: 340,
          padding: 15
        }}
        >

          <h1>CounterApp</h1>

          <h2>{ counter }</h2>

        <div>
          <button onClick={() => handleCounter(counter + 1)}>+1</button>
          <button onClick={() => handleCounter(counter + 10)}>x10</button>
          <button aria-label='btn-reset' onClick={() => handleCounter(value)}>Reset</button>
          <button onClick={() => handleCounter(counter - 1)}>-1</button>
          <button onClick={() => handleCounter(counter - 10)}>-10</button>
        </div>
      </div>
    </>
  );
};

Counter.propType = {
  value: number.isRequired
};

export default Counter;