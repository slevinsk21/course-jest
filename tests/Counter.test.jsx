import { fireEvent, render, screen } from '@testing-library/react';

import Counter from '../src/Counter';

describe('File: <Counter>', () => {
  test('#1: Test: ( must be do match to the snapshot )', () => {
    const { container } = render( <Counter value={ 0 } /> );
    expect( container ).toMatchSnapshot();
  });

  test('#2: Test: ( must be show initial value of 100 )', () => {
    const initialValue = 100;
    render( <Counter value={ initialValue } /> );
    expect( screen.getByText( initialValue ) ).toBeTruthy();
    expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain( `${initialValue}` );
  });

  test('#3 Test: ( must increment in 1 unity with the button ´+1´ )', () => {
    render( <Counter value={ 10 } /> );
    fireEvent.click( screen.getByText('+1') );
    expect( screen.getByText('11') ).toBeTruthy();
  });

  test('#4 Test: ( must decrement in 1 unity with the button ´-1´ )', () => {
    render( <Counter value={ 10 } /> );
    fireEvent.click( screen.getByText('-1') );
    expect( screen.getByText('9') ).toBeTruthy();
  });

  test('#5 Test: ( must reset the value the button ´reset´ )', () => {
    const initialValue = 3;
    render( <Counter value={ initialValue } /> );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );

    // click reset
    // fireEvent.click( screen.getByText( 'Reset' ) );
    fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );

    expect( screen.getByText( initialValue ) ).toBeTruthy();
  });
});
