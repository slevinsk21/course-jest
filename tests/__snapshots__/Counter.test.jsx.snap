// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: <Counter> #1: Test: ( must be do match to the snapshot ) 1`] = `
<div>
  <div
    style="height: 60px;"
  />
  <div
    style="background-color: rgba(221, 51, 221, 0.2); border-radius: 8px; display: flex; flex-direction: column; align-items: center; margin: 0px auto; max-width: 340px; padding: 15px;"
  >
    <h1>
      CounterApp
    </h1>
    <h2>
      0
    </h2>
    <div>
      <button>
        +1
      </button>
      <button>
        x10
      </button>
      <button
        aria-label="btn-reset"
      >
        Reset
      </button>
      <button>
        -1
      </button>
      <button>
        -10
      </button>
    </div>
  </div>
</div>
`;
